source/common/router/router.cc

```diff
@@ -1359,6 +1359,8 @@ void Filter::onUpstreamComplete(UpstreamRequest& upstream_request) {
                                              upstreamZone(upstream_request.upstreamHost())};
 
     code_stats.chargeResponseTiming(info);
+    cluster_->loadReportRouterStats().http_upstream_rq_time_.recordValue(
```

Comment:
I see, this is probably where we actually have been maintaining all the relevant upstream timers. 

We already do something like https://github.com/envoyproxy/envoy/blob/02a526257a013e80bc508399446aac0234c3ec4e/source/common/router/router.cc#L311 from the router to track stats. So, I think conceptually what you are doing is fine, you should just avoid naming the stats and interface as being router based, since what it really is is a property of the cluster.

Reply:
I'm going to change the name to `loadReportHistogramStats`, let me know if you have any better ideas